keps/sig-multicluster/1645-multi-cluster-services-api/README.md

```diff
@@ -758,6 +755,31 @@ _It is expected that the `.clusterset.local` zone is standard and available in
+
+##### Not allowing cluster-specific targeting via DNS
+
+While we reserve the form `<clusterid>.<svc>.<ns>.svc.clusterset.local.` for possible future use, both ClusterSetIP Services and Multicluster Headless Services are specified to explicitly disallow DNS records that target all 1+N backends in a single cluster.
+
+For ClusterSetIP services, this rationale is tied to the intent of its underlying ClusterIP Service. In a single-cluster setup, the purpose of a ClusterIP service is to reduce the context needed by the application to target ready backends, especially if those backends disappear or change frequently, and leverages kube-proxy to do this independent of the limitations of DNS. ([ref](https://kubernetes.io/docs/concepts/services-networking/service/#why-not-use-round-robin-dns)) Similarly, users of exported ClusterIP services should depend on the single `<clusterset-ip>` (or the single `A`/`AAAA` record mapped to it), instead of targeting per cluster backends. If a user has a need to target backends in a different way, they should use headless Services.
+
+For Multicluster Headless Services, the rationale is tied to the intent of its underlying Headless Service to provide absolutely no load balancing capabilities on any stateful dimension of the backends (such as cluster locality), and provide routing to each single backend for the application's purposes.
+
+In both cases, this restriction seeks to preserve the MCS position on [namespace sameness](https://github.com/kubernetes/community/blob/master/sig-multicluster/namespace-sameness-position-statement.md). Services of the same name/namespace exported in the multicluster environment are considered to be the same by definition, and thus their backends are safe to 'merge' at the clusterset level. If these backends need to be addressed differently based on other properties than name and namespace, they lose their fungible nature which the MCS API depends on. In these situations, those backends should instead be fronted by a Service with a different name and/or namespace.
+
+For example, say an application wishes to target the backends for a `ClusterSetIP ServiceExport` called `special/prod` in `<clusterid>=clusterEast` separately from all backends in `<clusterid>=clusterWest`. Instead of depending on the disallowed implementation of cluster-specific addressing, the Services in each specific cluster should actually be considered non-fungible and be created and exported by `ServiceExport`s with different names that honor the boundaries of their sameness, such as `specialEast/prod` for all the backends in `<clusterid>=clusterEast` and `specialWest/prod` for the backends in `<clusterid>=clusterWest`. In this situation, the resulting DNS names `specialEast.prod.svc.clusterset.local` and `specialWest.prod.svc.clusterset.local` encode the cluster-specific addressing required by virtue of being two different `ServiceExport`s.
```

Comment:
nit - capital letters are not allowed in service names :)

Reply:
Fixed in [c4e0549](https://github.com/kubernetes/enhancements/pull/2577/commits/c4e05493b6917d0fe1d8d451367259061eac704a)